packages/react-router/src/Match.tsx

```diff
@@ -127,6 +136,10 @@ function MatchInner({ matchId }: { matchId: string }): any {
     },
   })
 
+  const Comp = route.options.component ?? router.options.defaultComponent
+
+  const out = React.useMemo(() => (Comp ? <Comp /> : <Outlet />), [Comp])
```

Comment:
It may be worth checking out the benefit of using the `matchId` as the key on the `<Comp/>`.

Ultimately may not be what we are looking for if stuff like `useSearch` doesn't trigger a re-render, but might be worth checking out.

Reply:
like this?

```tsx
<Comp key={matchId}/> 
```

which effect would you expect from that?

Reply:
Yup. 

It's since keyed components don't re-render unless the key changes, it may help.

And we shouldn't have to worry about the render being stale, since the `matchId` changes if actual match info has changed.

Reply:
tried it with kitchen sink example, using the `matchId` as key causes the `InvoiceComponent` to be rendered only once instead of twice without it.
great find!

Reply:
🚀

Yeah it's a simple hack I tend to use to either trigger a child component's re-render or to prevent unneeded renders.